app/services/search/postgres/podcast_episode.rb

```diff
@@ -34,16 +34,26 @@ def self.search_documents(page: 0, per_page: DEFAULT_PER_PAGE, sort_by: nil, sor
         page = page.to_i + 1
         per_page = [(per_page || DEFAULT_PER_PAGE).to_i, MAX_PER_PAGE].min
 
-        relation = ::PodcastEpisode.includes(:podcast).available
+        relation = ::PodcastEpisode
+          .reachable
+          .where(podcast_id: Podcast.published)
+          .includes(:podcast)
+          .references(:podcasts)
```

Comment:
this is still going to select all the fields from `podcasts` :-P see https://gist.github.com/rhymes/9c624f476e773d05f693751c91209cf0

Rails will load the entire `podcasts` table so you might as well remove it from the `ATTRIBUTES` list:

```ruby
      ATTRIBUTES = [
        "body",
        "comments_count",
        "id",
        "podcast_id",
        "processed_html",
        "published_at",
        "quote",
        "reactions_count",
        "slug",
        "subtitle",
        "summary",
        "title",
        "website_url",
      ].freeze
```

see https://gist.github.com/rhymes/056a319fff31fcc2fb34c3c8d87c5855 for the difference in the query

Reply:
I can update it now!

Also, if I remove `references` it turns into 2 separate queries
```
  PodcastEpisode Load (3.5ms)  SELECT "podcast_episodes"."body", "podcast_episodes"."comments_count", "podcast_episodes"."id", "podcast_episodes"."podcast_id", "podcast_episodes"."processed_html", "podcast_episodes"."published_at", "podcast_episodes"."quote", "podcast_episodes"."reactions_count", "podcast_episodes"."slug", "podcast_episodes"."subtitle", "podcast_episodes"."summary", "podcast_episodes"."title", "podcast_episodes"."website_url" FROM "podcast_episodes" INNER JOIN (SELECT "podcast_episodes"."id" AS pg_search_id, (ts_rank((to_tsvector('simple', coalesce("podcast_episodes"."body"::text, '')) || to_tsvector('simple', coalesce("podcast_episodes"."subtitle"::text, '')) || to_tsvector('simple', coalesce("podcast_episodes"."title"::text, ''))), (to_tsquery('simple', ''' ' || 'python' || ' ''' || ':*')), 0)) AS rank FROM "podcast_episodes" WHERE ((to_tsvector('simple', coalesce("podcast_episodes"."body"::text, '')) || to_tsvector('simple', coalesce("podcast_episodes"."subtitle"::text, '')) || to_tsvector('simple', coalesce("podcast_episodes"."title"::text, ''))) @@ (to_tsquery('simple', ''' ' || 'python' || ' ''' || ':*')))) AS pg_search_1847632507b7f58950c333 ON "podcast_episodes"."id" = pg_search_1847632507b7f58950c333.pg_search_id WHERE "podcast_episodes"."reachable" = $1 AND "podcast_episodes"."podcast_id" IN (SELECT "podcasts"."id" FROM "podcasts" WHERE "podcasts"."published" = $2) LIMIT $3 OFFSET $4  [["reachable", true], ["published", true], ["LIMIT", 60], ["OFFSET", 0]]
  Podcast Load (0.4ms)  SELECT "podcasts".* FROM "podcasts" WHERE "podcasts"."id" IN ($1, $2, $3)  [[nil, 3], [nil, 2], [nil, 1]]
```

Locally, these combined are faster than the single query done when having `references`. What do you think? @rhymes @jgaskins

Reply:
_TL;DR: Just check the last sentence_

It's hard to say. Using `JOIN`s can be a bit slower for the DB but it can be _really_ rough on ActiveRecord because of how it chops up your result set. Apologies if this is explaining something you already know, but let's say you have a podcast with 3 episodes:

| podcast | podcast_episode |
|---------|-------------------|
| 1 | 1 |
| 1 | 2 |
| 1 | 3 |

Since SQL always returns data inside a table projection, a `JOIN`ed 1:N association will return the parent record for every single child record, so we get podcast `1` 3x in that result set. Rails chops this up so that, inside the application code, there's only one parent record (all the rest are discarded) and it contains an array of all the child records. This is an example of the [object-relational impedance mismatch](https://en.wikipedia.org/wiki/Object%E2%80%93relational_impedance_mismatch#Mapping_to_relational_concepts).

So using `includes` without `references` will be significantly easier on ActiveRecord if we aren't actually referencing the `podcast_episodes` records inside the query.

However, depending on the intent of the query, `includes` and `eager_load` (which is effectively just `includes.references`) could return different results. `includes` is similar to a `LEFT OUTER JOIN` since it will load all of the lefthand relation regardless of whether it matches with a record in the righthand relation. It's one of those distinctions I sorely wish I didn't have to know about. :joy: I think what we were discussing before was a left join, though, so if that's true and we aren't actually referencing `podcast_episodes` columns in the query, it should be perfectly fine.

Reply:
That makes a lot of sense, I really appreciate that explanation 💯 

I'm thinking we merge this as is then - we can tweak it later if needed. As is with the new index (:rocket:), the goal is accomplished.

Reply:
Great explanation @jgaskins!